front/components/assistant_builder/InstructionScreen.tsx

```diff
@@ -461,6 +510,65 @@ function AdvancedSettings({
   );
 }
 
+function PromptHistory({
+  history,
+  onVersionPick,
+}: {
+  history: LightAgentConfigurationType[];
+  onVersionPick: (
+    config: LightAgentConfigurationType,
+    isLatest: boolean
+  ) => void;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const latestConfig = history[0];
+  const [currentConfig, setCurrentConfig] =
+    useState<LightAgentConfigurationType>(latestConfig);
+
+  const getStringRepresentation = useCallback(
+    (config: LightAgentConfigurationType) => {
+      const dateAndHourMatch = config.versionCreatedAt?.match(
+        /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}/
+      )?.[0];
+
+      return config.version === latestConfig?.version
+        ? "Latest Version"
```

Comment:
I'd change to "Versions history" instead of "latest version" when currently on the latest version.

Reply:
Jumping back to the latest version after inspecting a previous version becomes confusing then, because you need to pick 'Version History'